InnerEye/ML/configs/other/fastmri_varnet.py

```diff
@@ -37,32 +43,131 @@ def log_image(self, name: str, image: torch.Tensor) -> None:
+                    test_path: str) -> LightningDataModule:
+    """
+    Creates a LightningDataModule that consumes data from the FastMRI challenge. The type of challenge
+    (single/multicoil) is determined from the name of the dataset in Azure blob storage. The mask type is set to
+    equispaced, with 4x acceleration.
+    :param azure_dataset_id: The name of the dataset (folder name in blob storage).
+    :param local_dataset: The local folder at which the dataset has been mounted or downloaded.
+    :param sample_rate: Fraction of slices of the training data split to use. Set to a value <1.0 for rapid prototyping.
+    :param test_path: The name of the folder inside the dataset that contains the test data.
+    :return: A LightningDataModule object.
+    """
+    if not azure_dataset_id:
+        raise ValueError("The azure_dataset_id argument must be provided.")
+    if not local_dataset:
+        raise ValueError("The local_dataset argument must be provided.")
+    for challenge in ["multicoil", "singlecoil"]:
+        if challenge in azure_dataset_id:
+            break
+    else:
+        raise ValueError(f"Unable to determine the value for the challenge field for this "
+                         f"dataset: {azure_dataset_id}")
+
+    mask = create_mask_for_mask_type(mask_type_str="equispaced",
+                                     center_fractions=[0.08],
+                                     accelerations=[4])
+    # use random masks for train transform, fixed masks for val transform
+    train_transform = VarNetDataTransform(mask_func=mask, use_seed=False)
+    val_transform = VarNetDataTransform(mask_func=mask)
+    test_transform = VarNetDataTransform()
+
+    return FastMriDataModule(data_path=local_dataset,
+                             test_path=local_dataset / test_path,
+                             challenge=challenge,
+                             sample_rate=sample_rate,
+                             train_transform=train_transform,
+                             val_transform=val_transform,
+                             test_transform=test_transform)
+
+
 class FastMri(LightningContainer):
+    """
+    A base class for all models for the FastMRI challenge. It implements the `create_model` overload, but not
+    the `get_data_module` method.
+    """
     # All fields that are declared here will be automatically available as commandline arguments.
-    challenge: str = param.String(default="multicoil", doc="Chooses between the singlecoil or multicoil"
-                                                           "acquisition setup.")
     sample_rate: Optional[float] = param.Number(default=None, doc="Fraction of slices of the training data split to "
                                                                   "use. Default: 1.0")
 
     def __init__(self) -> None:
         super().__init__()
-        self.azure_dataset_id = "fastmrimini_brain"
+        self.num_epochs = 50
```

Comment:
Hard coded on purpose?

Reply:
yes. The FB models are all trained with 50 epochs by default.